src/backend/timelineCredPlan.js

```diff
@@ -0,0 +1,33 @@
+//@flow
+
+import {Graph} from "../core/graph";
+import {TaskReporter} from "../util/taskReporter";
+import {type TimelineCredParameters} from "../analysis/timeline/params";
+import {type PluginDeclaration} from "../analysis/pluginDeclaration";
+import {TimelineCred} from "../analysis/timeline/timelineCred";
+
+type ComputeFn = (opts: {|
+  graph: Graph,
+  params?: $Shape<TimelineCredParameters>,
+  plugins: $ReadOnlyArray<PluginDeclaration>,
+|}) => Promise<TimelineCred>;
+
+export function timelineCredPlan(
+  pluginDeclarations: $ReadOnlyArray<PluginDeclaration>,
+  reporter: TaskReporter,
+  compute: ComputeFn
+) {
+  return async (
+    graph: Graph,
+    params: ?$Shape<TimelineCredParameters>
+  ): Promise<TimelineCred> => {
+    reporter.start("compute-cred");
+    const cred = await compute({
+      graph,
+      params: params || {},
```

Comment:
just pass thru params directly, `TimelineCred.compute` allows null/undefined params and then sets defaults.

Reply:
Not actually. What it allows is the `params` key to be missing from the object. Using null/undefined would cause a flow error here.

`params?: $Shape<TimelineCredParameters>` (from `TimelineCred.compute`)
versus
`params: ?$Shape<TimelineCredParameters>` (this function)